### PR TITLE
[WIP] More progress on sync

### PR DESCRIPTION
**Pull request by ErikBjare**
I wanted the aw-sync-rust implementation able to sync between both aw-server-python and aw-server-rust instances so I started working on the ability to use the REST API to get the job done.

### CODE REVIEW

**Comment by ErikBjare** [comment]
@johan-bjareholt I'm still in the middle of this PR, but could you look over the changes I've made so far and give me some feedback on my attempts to unify the datastore and aw-client-rust APIs?

I also heard you were working on a migration tool, and the things I'm building here could potentially give that for free.

File: aw-sync-rust/src/lib.rs [hunk 1]
```diff
@@ -6,14 +6,87 @@
...
 
 use chrono::{DateTime, Utc, Duration};
-use serde_json;
+use aw_client_rust::{AwClient};
+
+use aw_server::datastore::{Datastore, DatastoreError};
+use aw_server::models::{Event, Bucket};
+
+// This trait should be implemented by both AwClient and Datastore, unifying them under a single API
+pub trait AccessMethod: std::fmt::Debug {
```
Comment (johan-bjareholt) [inline aw-sync-rust/src/lib.rs]: AccessMethod is a very generic name and doesn't describe what it does

**Comment by ErikBjare** [comment]
@johan-bjareholt The latest commits here fixes aw-server-rust to work with the latest nightly and improves the build process on Android. 

Can we merge it since the WIP stuff it isolated in a member package anyway?

**Comment by johan-bjareholt** [comment]
@ErikBjare Seems like there is a merge conflict

> Can we merge it since the WIP stuff it isolated in a member package anyway?

Yes, thats fine IMO.

**Pull request merged**